src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/mgmt_vm/nested_templates/nsg_existing_or_none.json

```diff
@@ -2,6 +2,9 @@
   "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
   "contentVersion": "1.0.0.0",
   "parameters": {
+    "location": {
+      "type": "string"
+    },
```

Comment:
Isn't this why we have the azuredeploy_emtpy, so that this kind of thing isn't required? Should NSG reference a common template instead of a nested one?